server/router/router.go

```diff
@@ -320,6 +323,10 @@ func (router *router) setStopping(v bool) {
 	router.stopping = v
 }
 
+func (router *router) Done() <-chan bool {
```

Comment:
Router is an interface. And I think it's cleaner. Through a method. And we cannot allow anyone to put anything in the channel this way.